olive/data_container/container/base_container.py

```diff
@@ -0,0 +1,72 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from typing import ClassVar
+
+from pydantic import BaseModel
+
+from olive.data_container.config import DataContainerConfig, DefaultDataComponentCombos
+from olive.data_container.constants import DataContainerType, DefaultDataContainer
+from olive.data_container.registry import Registry
+
+
+@Registry.register(DataContainerType.DATA_CONTAINER, name=DefaultDataContainer.DATA_CONTAINER.value)
+class BaseContainer(BaseModel):
```

Comment:
>BaseContainer [](http://example.com/codeflow?start=6&length=13)

what's the relationship among dataset, dataloader and data container?

Reply:
Please refer to the design docs. Basically:
1. `data container` is the endpoints used to call the function under `data components`
2. `dataset/dataloader/pre_process/post_process` is the four data components.